build.sh

```diff
@@ -14,42 +14,37 @@ yarn workspace storage-node build
 yarn workspace pioneer build
 
 if ! command -v docker-compose &> /dev/null
-then
-  echo "docker-compose not found, skipping docker build!"
-else
-  # Build joystream/apps docker image
-  docker-compose build pioneer
-
-  # Optionally build joystream/node docker image
-  # TODO: Try to fetch a cached joystream/node image
-  # if one is found matching code shasum instead of building
-  while true
-  do
-    read -p "Rebuild joystream/node docker image? (y/N): " answer2
+  # Build or fetch cached joystream/node docker image
+  if [[ "$SKIP_JOYSTREAM_NODE" = 1 || "$SKIP_JOYSTREAM_NODE" = "true" ]]; then
+    echo "Skipping build of joystream/node docker image."
+  else
+    # Fetch a cached joystream/node image if one is found matching code shasum instead of building
+    CODE_HASH=`scripts/runtime-code-shasum.sh`
```

Comment:
I noticed that the hash that is calculated locally in my case differs from the one in the CI (locally it was `34e163be8e43c5631d8b92e9c43ab0bf0fa62b9c`)

When I ran just ran `./scripts/runtime-code-shasum.sh` from the command line I got a bunch of errors about unexisting files/directories (`Cargo.lock`, `runtime-modules`, `joystream-node.Dockerfile` etc.).

I traced it to:
```
export WORKSPACE_ROOT=`cargo metadata --offline --no-deps --format-version 1 | jq .workspace_root -r`
```
Which in my case returned `Command 'jq' not found`.

After installng `jq` the errors are no longer showing, but when I run `./scripts/runtime-code-shasum.sh` the hash is still different (`d63feb0531a22ffa2db1c04e63c7ff4fef3d3a97`)

Reply:
the difference may come if your version of `tar` is not gnu tar. The key thing for the hash to be deterministic is for the timestamps, ordering, ownership info of files to be calculated in the same way. What version of `tar` do you have?

Reply:
After some troubleshooting we came to realize the file mode can differ between systems so we added the `--mode` argument to the mix in https://github.com/Joystream/joystream/pull/2292/commits/f68b14f25b1dd48320b8f5671cd508a2018b3d51